apps/opik-backend/src/main/java/com/comet/opik/api/resources/v1/priv/CheckAccessResource.java

```diff
@@ -0,0 +1,59 @@
+package com.comet.opik.api.resources.v1.priv;
+
+import com.codahale.metrics.annotation.Timed;
+import com.comet.opik.api.ProjectNameHolder;
+import com.comet.opik.api.error.ErrorMessage;
+import com.comet.opik.domain.ProjectService;
+import com.comet.opik.infrastructure.auth.RequestContext;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.inject.Inject;
+import jakarta.inject.Provider;
+import jakarta.validation.Valid;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Path("/v1/private/check")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Timed
+@Slf4j
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+@Tag(name = "Check", description = "Access check resources")
+public class CheckAccessResource {
+
+    private final @NonNull ProjectService projectService;
+    private final @NonNull Provider<RequestContext> requestContext;
+
+    @POST
+    @Path("/project")
+    @Operation(operationId = "retrieveProject", summary = "Check user access to workspace and project", description = "Check user access to workspace and project", responses = {
```

Comment:
It does pretty much the same, but requirement was that we should just return ok/fail, without any payload, so existing endpoint could not be used.

Also since purpose for this endpoint is to check apiKey, workspace and project access, we could reuse existing service conveniently, but potentially we might need to check other combinations/entities, so it makes sense to separate it as a stand alone Resource.

Reply:
But there's no permission based on resources, just workspace. If the user has access to the workspace, it has access to all resources

Reply:
@thiagohora This is needed for SDK Dify integration. And project part is mandatory. 
Yes we don't have special permissions on project level, but this is checking that project belongs to workspace and user has access to that workspace.

Reply:
So, you want to check if the project exists and check the authentication and authorization, right? Why not call it 'v1/private/projects/exists'?

Reply:
There is no such endpoint `v1/private/projects/exists`. 

The closest is find all by name:
https://www.comet.com/docs/opik/reference/rest_api/find-projects
But it uses name as regexp, so it's not the same.

I agree that `retrieve` could be used here, let's raise this with the product, to make sure that they actually need new endpoint.

Reply:
As mentioned, a similar endpoint already exists:
- https://github.com/comet-ml/opik/blob/b601113826af9436eb241d91c97613bfe96f9857/apps/opik-backend/src/main/java/com/comet/opik/api/resources/v1/priv/ProjectsResource.java#L176-L193

If we were to create a new endpoint to check project existence and access, it should be under the Project resource, not here.

If the requirement was to not to delete any payload, that doesn't really matter as `Project` object is tiny, there won't be any difference from just checking the HTTP status code.

We should push back on low level requirements from product, specially if they cause maintainability issues and higher effort like this one.